src/components/hv-list/index.js

```diff
@@ -37,10 +46,100 @@ export default class HvList extends PureComponent<HvComponentProps, State> {
 
   props: HvComponentProps;
 
+  ref: ?ElementRef<typeof FlatList>;
+
   state: State = {
     refreshing: false,
   };
 
+  onRef = (ref: ?ElementRef<typeof FlatList>) => {
+    this.ref = ref;
+  };
+
+  onUpdate: HvComponentOnUpdate = (
+    href: ?DOMString,
+    action: ?DOMString,
+    element: Element,
+    options: HvComponentOptions,
+  ) => {
+    if (action === 'scroll' && options.behaviorElement) {
+      this.handleScrollBehavior(options.behaviorElement);
+      return;
+    }
+    this.props.onUpdate(href, action, element, options);
+  };
+
+  handleScrollBehavior = (behaviorElement: Element) => {
+    const targetId: ?DOMString = behaviorElement?.getAttribute('target');
+    if (!targetId) {
+      console.warn('[behaviors/scroll]: missing "target" attribute');
+      return;
+    }
+    const doc: ?Document = this.props.element.ownerDocument;
+    const targetElement: ?Element = doc?.getElementById(targetId);
+    if (!targetElement) {
+      return;
+    }
+
+    const targetElementParentList = getAncestorByTagName(
+      targetElement,
+      LOCAL_NAME.LIST,
+    );
+    if (targetElementParentList === this.props.element) {
```

Comment:
maybe return early in this case to remove a layer of nesting?

Reply:
[c5f912b](https://github.com/Instawork/hyperview/pull/669/commits/c5f912be897d590557aa44d2aa590d98ba7915a1)